internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -87,7 +119,19 @@ var secretSpec = &schema.Schema{
 	},
 }
 
-// nolint: unused
+func validateSpec(_ context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	switch {
+	case (diff.Get(helper.GetFirstElementOf(specKey, DockerConfigjsonKey, ImageRegistryURLKey))).(string) == "":
+		return fmt.Errorf("invalid value for %v, can not be nil string", ImageRegistryURLKey)
+	case (diff.Get(helper.GetFirstElementOf(specKey, DockerConfigjsonKey, UsernameKey))).(string) == "":
+		return fmt.Errorf("invalid value for %v, can not be nil string", UsernameKey)
+	case (diff.Get(helper.GetFirstElementOf(specKey, DockerConfigjsonKey, PasswordKey))).(string) == "":
+		return fmt.Errorf("invalid value for %v, can not be nil string", PasswordKey)
+	default:
+		return nil
+	}
+}
```

Comment:
Why is this needed? Can't we add schema validations instead?

Explore adding validation to the different blocks and attributes in the schema.

Reply:
removed